scripts/mostReadCollectionTimes.js

```diff
@@ -19,20 +20,38 @@ const services = Object.keys(allServices);
 
 console.log('services', services);
 
+const loadCerts = ({ caPath, certChainPath, keyPath }) =>
+  Promise.all([
+    fsPromises.readFile(caPath, 'UTF-8'),
+    fsPromises.readFile(certChainPath, 'UTF-8'),
+    fsPromises.readFile(keyPath, 'UTF-8'),
+  ]);
+
 const fetchWithCert = async (url, options) => {
-  const cert = fs.readFileSync(
-    process.env.CERT_CHAIN_PATH || '/etc/pki/tls/certs/client.crt',
-  );
-  const ca = fs.readFileSync(
-    process.env.CA_PATH || '/etc/pki/tls/certs/ca-bundle.crt',
-  );
-  const key = fs.readFileSync(
-    process.env.KEY_PATH || '/etc/pki/tls/private/client.key',
-  );
-  const agent = new Agent({ cert, ca, key });
-  return fetch(url, { agent, ...options });
-};
+  const caPath = process.env.CA_PATH || '/etc/pki/tls/certs/ca-bundle.crt';
+  const certChainPath =
+    process.env.CERT_CHAIN_PATH || '/etc/pki/tls/certs/client.crt';
+  const keyPath = process.env.KEY_PATH || '/etc/pki/tls/private/client.key';
+
+  const [ca, certChain, key] = await loadCerts({
+    caPath,
+    certChainPath,
+    keyPath,
+  });
 
+  return fetch(url, {
+    dispatcher: new Agent({
+      connect: {
+        secureContext: createSecureContext({
+          cert: certChain,
+          key,
+          ca,
+        }),
+      },
+    }),
+    ...options,
```

Comment:
Is this just a nice refactor or does the new native fetch need this new 'dispatcher' and 'connect' to work ?

Reply:
This is required change as we need to read in the certs async instead of sync. It more closely aligns the pattern we have in our main `getCerts` file that we use for the application requests.

Reply:
Good spot, shall remove these!